textfrag/src/types/directive/fragdirective.rs

```diff
@@ -0,0 +1,336 @@
+/// Fragment Directive object is collection of text fragments in the URL's fragment
+/// :~: is the fragment directive delmiter, followed by list of `[TextDirective]`
+use std::{cell::RefCell, collections::HashMap};
+
+use html5ever::tokenizer::{BufferQueue, Tokenizer, TokenizerOpts};
+use url::Url;
+
+use crate::{
+    extract::FragmentDirectiveTokenizer,
+    types::{
+        FragmentDirectiveError, FragmentDirectiveStatus, TextDirective, TextDirectiveStatus,
+        TextFragmentError,
+    },
+};
+
+pub const FRAGMENT_DIRECTIVE_DELIMITER: &str = ":~:";
+
+/// Fragment Directive defines the base url and collection of Text Directives
+#[derive(Default, Clone, Debug)]
+pub struct FragmentDirective {
+    #[allow(dead_code)]
+    url: Option<Url>,
+    text_directives: RefCell<Vec<TextDirective>>,
+}
+
+impl FragmentDirective {
+    pub fn text_directives(&self) -> Vec<TextDirective> {
+        self.text_directives.borrow().to_owned()
+    }
+
+    pub fn text_directives_mut(&self) -> Vec<TextDirective> {
+        self.text_directives.borrow_mut().to_owned()
+    }
+
+    // Extract Text Directives, from the Url fragment string
```

Comment:
Use three slashes for function/method comments. This way they show up in the docs later. 😎 

```suggestion
    /// Extract Text Directives, from the Url fragment string
```